load.go

```diff
@@ -335,8 +342,30 @@ func (c *Command) processBuildArgs(lex *shell.Lex, args map[string]string, name
 	return nil
 }
 
+func (s Spec) FillDefaults() Spec {
+	for _, src := range s.Sources {
+		fillDefaults(&src)
+	}
+
+	for k, patches := range s.Patches {
+		for i, ps := range patches {
+			if ps.Strip != nil {
+				continue
+			}
+			strip := DefaultPatchStrip
+			s.Patches[k][i].Strip = &strip
+		}
+	}
+
+	return s
+}
+
 func (s Spec) Validate() error {
```

Comment:
After splitting out the build-arg substitution logic from spec loading, I think it makes sense to split out the validation and default filling logic too (there was a bug with duplicate validation and default filling)